internal/state/state.go

```diff
@@ -69,3 +135,39 @@ func (state *State) RunGC(ctx context.Context, logger *zap.Logger, interval time
 		}
 	}
 }
+
+// Describe implements prom.Collector interface.
+func (state *State) Describe(ch chan<- *prom.Desc) {
+	prom.DescribeByCollect(state, ch)
+}
+
+// Collect implements prom.Collector interface.
+func (state *State) Collect(ch chan<- prom.Metric) {
+	state.rw.RLock()
+	defer state.rw.RUnlock()
+
+	var mAffiliates, mEndpoints, mSubscriptions int
+	for _, c := range state.clusters {
+		mAffiliates += len(c.affiliates)
```

Comment:
this is not safe: accessing cluster state without a lock is not safe.

and I think those metrics should be collected on the fly (that is, we should track number of affiliates/clusters as they get updated instead of trying to walk whole state).